### PR TITLE
Add build system + build on GH Actions

### PR DESCRIPTION
**Pull request by k-mack**
This is prerequisite work for #20.

Before we can deploy the app to any host provider, we need to be able to build a production distribution of the application.
Instead of using `create-react-app`/`webpack`, I choose [Vite](https://vitejs.dev/) since it's smaller and faster.

To conform to Vite's usage, I had to move `index.html` to the root directory. Also, Vite [does not allow JSX syntax in JS files](https://github.com/vitejs/vite/issues/2727), so I had to change the file extensions of the `.js` files that use JSX to `.jsx`.

I added a GH workflow to build the code upon any push.
This begins to lay the foundation for additional CI workflows, namely to deploy to GH Pages.

### CODE REVIEW

**Comment by k-mack** [comment]
This can wait until #30 is merged.

File: docs/README.md
Comment (leekahung) [inline docs/README.md]: Can change port 1234 to port 5173 on Line 53 for README.md update along with "Parcel v2" on Line 17 with "Vite".

After those, rebuild the JSDocs documentation with npm run docs and I think we're good with the changes after #30 

Got everything running just fine on my local computer.

**Review by leekahung — CHANGES_REQUESTED** [review]

File: docs/README.md
Comment (leekahung) [reply docs/README.md:53]: Can also change "Parcel v2" on Line 17 with "Vite" in update.

Comment (leekahung) [reply docs/README.md]: Gonna revise changes via PR. I'll close this conversation.

**Review by leekahung — APPROVED** [review]

**Comment by leekahung** [comment]
I've made a few minor changes to the docs/README.md and regenerated the JSDocs on #32 

Other than that, I'm good with the changes! Thanks Kevin!

**Comment by k-mack** [comment]
@leekahung I can certainly pull in your changes: they are excellent suggestions.

WDYT about removing the generated docs in a follow-up PR? I'm not used to committing generated documentation to Git ... it can usually cause a lot of churn within the repository.

Does it make sense to add the generated docs to `.gitignore` and document in `docs/README.md` how a developer can generate the docs for themselves? If we do that, then we'll remove the burden of always having to commit the latest generated docs into the repository when we make code changes.

**Comment by leekahung** [comment]
@k-mack Yeah, that's certainly doable. The Node script to generate docs is already in package.json, so future users just need to run `npm run docs` in the root directory and to run 
`npx serve docs` to view the them. So a comment somewhere in the existing documentation (outside of docs) would be nice. I think somewhere inside CONTRIBUTING.md would be a good idea. I'll have a look.

**Comment by leekahung** [comment]
Just made the changes in #33 

A summary on how this would work, would be:
Danica's CONTRIBUTING.md will link to the last remaining file in /docs (/docs/README.md), which will provide instructions on how Users/Developer can generate their own React docs locally.

Then .gitignore will also prevent any generated files/directories made from JSDocs on their local computer to GitHub.

A clean-up script was also made if users/developers want to clear their local /docs directory. Instructions are also provided inside /docs/README.md

**Comment by leekahung** [comment]
Hey @k-mack, @Jared-Krajewski, @danicaboe, @GoodGuyGregory, since #30 has yet to be pushed into Master and the changes there is mainly focused on CONTRIBUTING.md and README.md, what do you think about merging #31 to Development before #30 is merged to Master?

At the moment, #30 follows the previous build with Parcel and the full React documentation is flushed out on GitHub.

With #31, we've switched to Vite and updated the application files to fit the new bundler. We've also clean up the React documentation in way that Users/Dvelopers can still access them by following the instructions from CONTRIBUTING.md in #30 and the updated /docs/README.md in #31, but not bloat what we have on GitHub.

If we merge #31 now with Development, all these changes will be reflected on Master when #30 is completed.

We can keep CONTRIBUTING.md and README.md being modified on #30, keep what we have here on #31 and modify the package.json on #31 to the correct version and license.

**Comment by k-mack** [comment]
I'd be fine with merging now. Having Vite in Master will make it easier for newcomers to get started with the project. Let's see if the others chime in. We can resolve this during Tuesday's meeting if need be.

**Review by leekahung — APPROVED** [review]
I'm in approval as of the latest commit here. That includes changes to CONTRIBUTING.md, READEME.md, and package.json to match #30.

**Comment by k-mack** [comment]
@leekahung are you sure your most recent commit was necessary. I believe this PR should have been able to be cleanly and non-destructively merged without it.

**Comment by leekahung** [comment]
@k-mack Would it not affect the merge? If that's the case, I'm okay with reverting the changes. I just wasn't sure.

**Comment by k-mack** [comment]
I tested it locally:

```bash
git fetch origin
git checkout -b test 29eddab
git checkout -b test-dev origin/Development
git merge test
```

I don't see any issues.

Mind rebasing the branch and dropping the commit?

**Comment by leekahung** [comment]
You got it! I've just reverted the branch to before the commit.

**Review by Jared-Krajewski — APPROVED** [review]
looks good to merge with #30 and send to master.

**Comment by k-mack** [comment]
2/3 is good for me.

**Pull request merged**
